### PR TITLE
Employers scripts update for experiments

### PR DESCRIPTION
**Pull request by chama-ram**
Updated classes of phone number input fields and targeting the elements more specifically. 

This is to allow the experiment to run smoothly while have various forms in place. In order to set up the A/B Test in Google optimize, we need to make changes to the script to avoid any elements from clashing

### CODE REVIEW

File: offerzen/global/employers/employers@5.5A.js [hunk 1]
```diff
@@ -0,0 +1,291 @@
...
+        }
+
+        return !!match;
+      });
+    }
+
+    function showFullListofTechRoles() {
+      showMoreLink = $('.js-show-more');
+      faidingList = $('.fading-list');
+      fullList = $('.full-list');
```
Comment (jethroflanagan) [inline offerzen/global/employers/employers@5.5A.js:199]: Please scope

File: offerzen/global/employers/employers@5.5A.js [hunk 2]
```diff
@@ -0,0 +1,291 @@
...
+        e.preventDefault();
+        e.stopImmediatePropagation();
+        faidingList.hide();
+        fullList.show();
+        showMoreLink.hide();
+      });
+    }
+
+    function matchCheckboxStates() {
+      $('.w-checkbox').each(function () {
```
Comment (jethroflanagan) [inline offerzen/global/employers/employers@5.5A.js:211]: Please scope

File: offerzen/global/employers/employers@5.5A.js [hunk 3]
```diff
@@ -0,0 +1,291 @@
...
+        const el = $(this);
+        const inputField = el.find('.w-checkbox-input');
+        if (el.find('input[type=checkbox]').is(':checked')) {
```
Comment (jethroflanagan) [inline offerzen/global/employers/employers@5.5A.js:214]: These are fine to leave once above version is scoped

File: offerzen/global/employers/employers@5.5A.js [hunk 4]
```diff
@@ -0,0 +1,291 @@
...
+      //Check Recaptcha error
+      const urlParams = new URLSearchParams(window.location.search);
+      if (urlParams.has('r')) {
+        $('.recaptcha-error').show();
+      }
+
+      $('#in-office-checkbox, #fully-remote-checkbox, #hybrid-checkbox').on(
+        'change',
+        function (e) {
+          let form = $('#wf-Company-Lead-Form');
```
Comment (jethroflanagan) [inline offerzen/global/employers/employers@5.5A.js:236]: Should this be `#og-company-signup-form`?

File: offerzen/global/employers/employers@5.5A.js [hunk 5]
```diff
@@ -0,0 +1,291 @@
...
+        event: event || 'Company Lead Form Submitted',
+        action: action || 'Lead Form Submitted',
+        label: label || 'Company Sign Up / Employer Landing Page',
+        category: category || 'Core',
+        source: source || 'Demand Sign Up',
+      });
+    }
+
+    function onSubmitCompanyLeadForm(token, e) {
+      const form = $('#wf-Company-Lead-Form');
```
Comment (jethroflanagan) [inline offerzen/global/employers/employers@5.5A.js:58]: Check scoping

File: offerzen/global/employers/employers@5.5A.js [hunk 6]
```diff
@@ -0,0 +1,291 @@
...
+    // ------------------------------------------------------------
+
+    (function init() {
+      //Check Recaptcha error
+      const urlParams = new URLSearchParams(window.location.search);
+      if (urlParams.has('r')) {
+        $('.recaptcha-error').show();
+      }
+
+      $('#in-office-checkbox, #fully-remote-checkbox, #hybrid-checkbox').on(
```
Comment (jethroflanagan) [inline offerzen/global/employers/employers@5.5A.js:233]: Please scope

File: offerzen/global/employers/employers@5.5A.js [hunk 7]
```diff
@@ -0,0 +1,291 @@
...
+      let isPhoneReady = false;
+
+      function onFormReady() {
+        if (!isPhoneReady || !isFormReady) return;
+
+        // Form is ready, but might still be waiting for recaptcha. That has it's own check in submit, so can be ignored
+        const submitButton = $('#wf-Company-Lead-Form input[type=submit]');
+        submitButton.attr('disabled', false);
+
+        $('#wf-Company-Lead-Form').on('submit', function (e) {
```
Comment (jethroflanagan) [inline offerzen/global/employers/employers@5.5A.js:259]: Check scoping

File: offerzen/global/employers/employers@5.5A.js [hunk 8]
```diff
@@ -0,0 +1,291 @@
...
+
+      // Load testing
+      let isFormReady = false;
+      let isPhoneReady = false;
+
+      function onFormReady() {
+        if (!isPhoneReady || !isFormReady) return;
+
+        // Form is ready, but might still be waiting for recaptcha. That has it's own check in submit, so can be ignored
+        const submitButton = $('#wf-Company-Lead-Form input[type=submit]');
```
Comment (jethroflanagan) [inline offerzen/global/employers/employers@5.5A.js:256]: Check scoping

File: offerzen/global/employers/employers@5.5B.js [hunk 9]
```diff
@@ -0,0 +1,288 @@
...
+
+      return roleTypes;
+    }
+
+    function trackSubmission() {
+      var event = $('.js-analytics-event').text();
+      var action = $('.js-analytics-action').text();
+      var label = $('.js-analytics-label').text();
+      var category = $('.js-analytics-category').text();
+      var source = $('.js-analytics-source').text();
```
Comment (jethroflanagan) [inline offerzen/global/employers/employers@5.5B.js:46]: Please scope

File: offerzen/global/employers/employers@5.5B.js [hunk 10]
```diff
@@ -0,0 +1,288 @@
...
+
+  window.$parsleyLoaded(function (window, document, parsley) {
+    function getRoleTypes(formProperties) {
+      let fields = [
+        'role_type[In-office]',
+        'role_type[Fully-remote]',
+        'role_type[Hybrid]',
+      ];
+      let roleTypes = [];
+      $('.js-val-msg-roles').hide();
```
Comment (jethroflanagan) [inline offerzen/global/employers/employers@5.5B.js:20]: Please scope

File: offerzen/global/employers/employers@5.5B.js [hunk 11]
```diff
@@ -0,0 +1,288 @@
...
+              key.replace('role_type[', '').replace(']', '').replace('-', ' ')
+            );
+          } else {
+            roleTypes.push(key.replace('role_type[', '').replace(']', ''));
+          }
+          delete formProperties[key];
+        }
+      }
+      if (roleTypes.length < 1) {
+        $('.js-val-msg-roles').show();
```
Comment (jethroflanagan) [inline offerzen/global/employers/employers@5.5B.js:35]: Please scope

**Review by jethroflanagan — CHANGES_REQUESTED** [review]
There are a number of elements that could be problematic if not scoped.

Recommend 2 "global" variables in the top of the function, e.g. `const ogForm =...` then use `ogForm.find('x')` and add another for multistep form

File: offerzen/global/employers/employers@5.5A.js (hunk 1 above)
Comment (chama-ram) [reply offerzen/global/employers/employers@5.5A.js:210]: This is not part of the form

**Review by jethroflanagan — APPROVED** [review]

**Review by michael616kriel — APPROVED** [review]

**Pull request merged**